### DISCUSSION TITLE
Can we use 24bit/192?

### DISCUSSION THREAD

**Discussion by Javimetal** [pschatzmann/arduino-audio-tools#367] [Q&A] [answered]
By using ESP32 or STM32, I'm interested in building a 24/192 audio effects device.

I know this library can be used with ESP32, but I am not sure about audio resolution.

**Comment by pschatzmann** [comment, marked as answer]
I don't really have any experience with this: With I2S, I am usually using 16.bits and never go over 44100hz.
But up to 32 bits are supported and if you use Google you can find that the maximum stable sample frequency appears to be 1.3 MHz.

I would expect that you might run into calculation perfmance issues at that speed: Int16 operations are fast but int32 and floating point operations get much slower!

The ESP32 might be on par (or slightly faster) with an STM32F4 (e.g.Black Pill). 
I am wondering how it compares with a STM32F7...

**Reply by Javimetal** [reply]
![Screenshot_20220920-141856_Chrome](https://user-images.githubusercontent.com/6794714/191255889-6b05ced9-3356-4aeb-9ecf-66f57c6d27c7.jpg)
Well, I'll have a try.

Thanks

**Comment by Javimetal** [comment]
![Error_Arduino_IDE2 0](https://user-images.githubusercontent.com/6794714/191670629-6380fb3c-333e-4319-a2ad-570bd3c2ed59.png)
Do you know why can't compile subfolders? (Arduino IDE 2.0)

**Comment by pschatzmann** [comment]
I guess because you forgot to install the related codec library
https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio
